### PR TITLE
[libc][math] Add C23 math function frexpf128.

### PR DESCRIPTION
**Pull request by lntue**

### CODE REVIEW

File: libc/test/src/math/smoke/FrexpTest.h [hunk 1]
```diff
@@ -93,4 +88,5 @@ template <typename T> class FrexpTest : public LIBC_NAMESPACE::testing::Test {
   using LlvmLibcFrexpTest = FrexpTest<T>;                                      \
   TEST_F(LlvmLibcFrexpTest, SpecialNumbers) { testSpecialNumbers(&func); }     \
   TEST_F(LlvmLibcFrexpTest, PowersOfTwo) { testPowersOfTwo(&func); }           \
-  TEST_F(LlvmLibcFrexpTest, SomeIntegers) { testSomeIntegers(&func); }
+  TEST_F(LlvmLibcFrexpTest, SomeIntegers) { testSomeIntegers(&func); }         \
+  static_assert(true, "Require semicolon.")
```
Comment (nickdesaulniers) [inline libc/test/src/math/smoke/FrexpTest.h:92]: lol

**Review by nickdesaulniers — APPROVED** [review]
idk about that static_assert; I don't have ideas on what you could replace it with, but it feels perhaps excessive to add.  In particular, I worry if others will start adding that all over the place (hopefully not).

**Comment by lntue** [comment]
> idk about that static_assert; I don't have ideas on what you could replace it with, but it feels perhaps excessive to add. In particular, I worry if others will start adding that all over the place (hopefully not).

I couldn't find something better for requiring semicolon when it's not inside a function body either.  But it makes the formatter much happier.  I'm aiming to change the `DECLARE_SPECIAL_CONSTANTS` macro so that it won't have to sit oddly separated from surrounding.

**Comment by jplehr** [comment]
This broke the AMDGPU OpenMP / GPU-libc buildbot (https://lab.llvm.org/staging/#/builders/11/builds/1921)
@jhuber6 can you please have a look at that?
Thanks!

**Comment by jplehr** [comment]
> This broke the AMDGPU OpenMP / GPU-libc buildbot (https://lab.llvm.org/staging/#/builders/11/builds/1921) @jhuber6 can you please have a look at that? Thanks!

I believe the broken build was fixed in https://github.com/llvm/llvm-project/pull/81438
Thanks!

**Pull request merged**